clients/src/main/java/org/apache/kafka/common/requests/ApiVersionsRequest.java

```diff
@@ -41,13 +42,30 @@ public Builder() {
             super(ApiKeys.API_VERSIONS);
         }
 
+        public Builder(String clientName) {
+            super(ApiKeys.API_VERSIONS);
+            customizedClientName = clientName;
+        }
+
         public Builder(short version) {
             super(ApiKeys.API_VERSIONS, version);
         }
 
+        public Builder(short version, String clientName) {
+            super(ApiKeys.API_VERSIONS, version);
+            customizedClientName = clientName;
+        }
+
         @Override
         public ApiVersionsRequest build(short version) {
-            return new ApiVersionsRequest(DATA, version);
+            if (customizedClientName == null) {
+                return new ApiVersionsRequest(DATA, version);
+            } else {
+                ApiVersionsRequestData customizedData = new ApiVersionsRequestData()
+                    .setClientSoftwareName(customizedClientName)
+                    .setClientSoftwareVersion(AppInfoParser.getVersion());
```

Comment:
I think this will get us the open source Kafka version? For LKC, I think we'd like to have a lkc version, instead of the base Kafka vesion.

Reply:
wonder if we can get a commit hash or tag name ...